### ISSUE TITLE
Delete Patch throws error "Cannot read properties of undefined"

### ISSUE THREAD

**Issue by svdvonde** [onsetsoftware/automerge-patcher#5]
The following program throws an error `TypeError: Cannot read properties of undefined (reading 'deleteAt')` when trying to patch an Automerge document.

```typescript
import {next as Automerge} from "@automerge/automerge"
import * as AutomergePatcher from "@onsetsoftware/automerge-patcher";

const doc = Automerge.from({x: 1});
Automerge.change(doc, (d) => {
    AutomergePatcher.patch(d, {
        action: "del",
        path: ["x"]
    });
});
```

The patch that I am trying to apply would be the result of the following `change` expression:
```
Automerge.change(doc, (d) => {
    delete d.x;
});
```

I am using the following versions:

 * `@automerge/automerge`: `2.1.3`
 * `@onsetsoftware/automerge-patcher`: `0.8.0`

**Comment by acurrieclark** [comment]
Interesting, so you are essentially trying to delete a top level property?

**Comment by svdvonde** [comment]
Correct - it came up when playing with some code/examples. I figured that it's an innocent operation that should work - but stumbled on this (I assume) edge case.

**Comment by acurrieclark** [comment]
Thanks for this. One which slipped through the net.

This should be closed by b8a881820160102f6ebc8bcd2a3db905506304ae. v0.8.1 has just been released with the fix.

Will close for now, but please feel free to reopen if this doesn't solve your issue.

**Comment by svdvonde** [comment]
Confirmed to be working in my examples that use `automerge-patcher`. Thank you!
